pkg/model/api/v1/secret.go

```diff
@@ -25,6 +24,8 @@ type SecretSpec struct {
 	BasicAuth *secret.BasicAuth `json:"basicAuth,omitempty" yaml:"basicAuth,omitempty"`
 	// The HTTP authorization credentials for the targets.
 	Authorization *secret.Authorization `json:"authorization,omitempty" yaml:"authorization,omitempty"`
+	// Oauth configuration for the targets.
+	OAuth *secret.OAuth
```

Comment:
```suggestion
OAuth *secret.OAuth `json:"oauth,omitempty" yaml:"oauth,omitempty"`
```

Reply:
And since you are updating the `SecretSpec`, can you also update the `Encrypt` and `Decrypt` function here: https://github.com/perses/perses/blob/main/internal/api/crypto/crypto.go#L65

Any secret field identified by the usage of `secret.Hidden` when it goes to public, should be encrypted